src/components/MultiHopTrade/hooks/useIsTradingActive.ts

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from 'react'
+import { getIsTradingActiveApi } from 'state/apis/swapper/getIsTradingActiveApi'
+import { selectActiveSwapperName } from 'state/slices/tradeQuoteSlice/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+import { selectBuyAsset, selectSellAsset } from 'state/zustand/swapperStore/selectors'
+import { useSwapperStore } from 'state/zustand/swapperStore/useSwapperStore'
+
+export const useIsTradingActive = () => {
+  const [isTradingActiveOnSellPool, setIsTradingActiveOnSellPool] = useState(false)
+  const [isTradingActiveOnBuyPool, setIsTradingActiveOnBuyPool] = useState(false)
+
+  const dispatch = useAppDispatch()
+
+  const buyAsset = useSwapperStore(selectBuyAsset)
+  const sellAsset = useSwapperStore(selectSellAsset)
+  const sellAssetId = sellAsset?.assetId
+  const buyAssetId = buyAsset?.assetId
+
+  const { getIsTradingActive } = getIsTradingActiveApi.endpoints
+  const swapperName = useAppSelector(selectActiveSwapperName)
+
+  useEffect(() => {
+    ;(async () => {
+      const isTradingActiveOnSellPoolResult =
+        sellAssetId &&
+        swapperName &&
+        (
+          await dispatch(
+            getIsTradingActive.initiate({
+              assetId: sellAssetId,
+              swapperName,
+            }),
+          )
+        ).data
+
+      const isTradingActiveOnBuyPoolResult =
+        buyAssetId &&
+        swapperName &&
+        (
+          await dispatch(
+            getIsTradingActive.initiate({
+              assetId: buyAssetId,
+              swapperName,
+            }),
+          )
+        ).data
+
+      setIsTradingActiveOnSellPool(!!isTradingActiveOnSellPoolResult)
+      setIsTradingActiveOnBuyPool(!!isTradingActiveOnBuyPoolResult)
+    })()
+  }, [swapperName, buyAssetId, dispatch, getIsTradingActive, sellAssetId])
```

Comment:
could be useful to split this useEffect into 2 - one for buy and one for sell such that changing buy asset doesnt trigger a fetch on the sell endpoint and vice versa

Reply:
I like it: https://github.com/shapeshift/web/pull/4889/commits/3c2dfb0db4a92e0f1a79c3401a5342d15ab69a42